src/wp-includes/comment-template.php

```diff
@@ -2006,22 +2016,56 @@ function comment_form_title( $no_reply_text = false, $reply_text = false, $link_
 		$reply_text = __( 'Leave a Reply to %s' );
 	}
 
-	$reply_to_id = isset( $_GET['replytocom'] ) ? (int) $_GET['replytocom'] : 0;
+	if ( 0 === $post_id ) {
+		echo $no_reply_text;
+		return;
+	}
+
+	$reply_to_id = _get_comment_reply_id( $post_id );
 
-	if ( 0 == $reply_to_id ) {
+	if ( 0 === $reply_to_id ) {
 		echo $no_reply_text;
+		return;
+	}
+
+	if ( $link_to_parent ) {
+		$author = '<a href="#comment-' . get_comment_ID() . '">' . get_comment_author( $reply_to_id ) . '</a>';
 	} else {
-		// Sets the global so that template tags can be used in the comment form.
-		$comment = get_comment( $reply_to_id );
+		$author = get_comment_author( $reply_to_id );
+	}
 
-		if ( $link_to_parent ) {
-			$author = '<a href="#comment-' . get_comment_ID() . '">' . get_comment_author( $comment ) . '</a>';
-		} else {
-			$author = get_comment_author( $comment );
-		}
+	printf( $reply_text, $author );
+}
 
-		printf( $reply_text, $author );
+/**
+ * Gets the comment's reply to ID from the $_GET['replytocom'].
+ *
+ * @since 5.9.0
+ * @access private
+ *
+ * @param int $post_id Post ID.
+ * @return int Comment's reply to ID.
+ */
+function _get_comment_reply_id( $post_id ) {
```

Comment:
@costdev What if the `$post_id` is not an integer but instead is an instance of `WP_Post`? This can happen in `comment_form()` which accepts either an `int` or `WP_Post` for its `$post_id` which is then passed through to `get_comment_id_fields( $post_id )` which then passes it to `_get_comment_reply_id( $post_id  )`.

Reply:
Right you are! I can add this in the next iteration:

```php
if ( $post_id instanceof WP_Post ) {
    $post_id = $post_id->ID;
}
```
**Confidence check request**

In the context of an extender using `comment_form()` or other methods that invoke `_get_comment_reply_id()`, if `$post_id` can be a `WP_Post` object, that means `$post_id` _could_ come from any method of getting a `WP_Post` object, including [wp_insert_post()](https://developer.wordpress.org/reference/functions/wp_insert_post/) et al.  This means that it _could_ also be a `WP_Error` object, or any other non-`int`, non-`WP_Post` object.

We can expect that all extenders, core contributors and such add their own validation against this, or we could guard against it directly in `_get_comment_reply_id()`.

With the `WP_Post` check above, we can now just guard against any `$post_id` value that isn't an `int`.

```diff
if ( ! $comment instanceof WP_Comment ||
	0 === (int) $comment->comment_approved ||
+	! is_int( $post_id ) ||
	$post_id !== (int) $comment->comment_post_ID ) {
		return 0;
}
```

Tests will need to be updated to cover whichever changes are agreed on.

**FYI**
The tests definitely need some additional refactoring before this is ready for `commit`. I realised that the data providers are using `$this`/`self`. Oops.

Reply:
As pseudo code, core does the following for functions that accept either a post ID or a post object:

```php
function fn( WP_Post|Int $post /* = null if default ought to be global post within the loop */ ) {
  $post = get_post( $post );
  if ( ! $post ) {
    /* Some fail safe, often a return */
  }
  $post_id = $post->ID;
}
```

It might be worth doing this throughout as most functions that accept a post object reference do so as either an integer or post object.

Reply:
Not sure if we have consensus on the path forward. Could I get some input from both on this? 🙂

Reply:
Note [74768e8](https://github.com/WordPress/wordpress-develop/pull/1607/commits/74768e85812a914a04e89feb26aa394e2f94e3cb) which I _think_ does what we were talking about. But again, please clarify 😅

Reply:
Sorry for the massive delays in reviewing this.

I agree with @peterwilsoncc to use Core's `get_post()` example as it is prevalent throughout Core.

@costdev notice `get_post()` is used in this file as well (e.g. `get_comments_number()`, `comments_open()`, `pings_open()`, etc.).

Beyond reusing similar coding patterns, there's another good reason to pass it into `get_post()`:

Verifies the post exists. If it doesn't exist, `null` is returned.